src/debugger/nodeDebugWrapper.ts

```diff
@@ -1,194 +1,246 @@
+                        } else {
+                            return Q({})
+                                .then(() => {
+                                    generator.step("checkPlatformCompatibility");
+                                    TargetPlatformHelper.checkTargetPlatformSupport(nodeDebugWrapper.mobilePlatformOptions.platform);
+                                    generator.step("startPackager");
+                                    Log.logMessage("Starting React Native Packager.");
+                                    return nodeDebugWrapper.remoteExtension.startPackager();
+                                })
+                                .then(() => {
+                                    // We've seen that if we don't prewarm the bundle cache, the app fails on the first attempt to connect to the debugger logic
+                                    // and the user needs to Reload JS manually. We prewarm it to prevent that issue
+                                    generator.step("prewarmBundleCache");
+                                    Log.logMessage("Prewarming bundle cache. This may take a while ...");
+                                    return nodeDebugWrapper.remoteExtension.prewarmBundleCache(nodeDebugWrapper.mobilePlatformOptions.platform);
+                                })
+                                .then(() => {
+                                    generator.step("mobilePlatform.runApp");
+                                    Log.logMessage("Building and running application.");
+                                    return mobilePlatform.runApp();
+                                })
+                                .then(() =>
+                                    nodeDebugWrapper.attachRequest(originalRequest, this, request, args, mobilePlatform));
+                        }
+                    }).catch(error =>
+                        nodeDebugWrapper.bailOut(this, error.message));
             });
+        };
+    }
 
-            // We do not permit arbitrary args to be passed to our process
-            args.args = [
-                args.platform,
-                debugServerListeningPort.toString(),
-                !isNullOrUndefined(args.iosRelativeProjectPath) ? args.iosRelativeProjectPath : IOSPlatform.DEFAULT_IOS_PROJECT_RELATIVE_PATH,
-                args.target || "simulator",
-            ];
+    /**
+     * Intecept the "attachRequest" instance method of NodeDebugSession to interpret arguments
+     */
+    private customizeAttachRequest(): void {
+        const nodeDebugWrapper = this;
+        this.nodeDebugSession.prototype.attachRequest = function (request: any, args: IAttachRequestArgs) {
+            nodeDebugWrapper.requestSetup(this, args);
+            nodeDebugWrapper.attachRequest(nodeDebugWrapper.nodeDebugSession.prototype.launchRequest, this, request, args, null);

```

Comment:
Will `nodeDebugWrapper.nodeDebugSession.prototype.launchRequest` be guaranteed to be the original launch request, or could it already be replaced by us? Perhaps we should grab references to launchRequest early on at global scope?